docs/authentication.md

```diff
@@ -0,0 +1,142 @@
+Authentication
```

Comment:
The only thing style-wise is that, since we don't know what markdown parser will be used, I would stick to atx-style headers, e.g. `#Authentication`, `##Auth flow`